partner_mobile_duplicate_warn/models/res_partner.py

```diff
@@ -15,26 +16,44 @@ class ResPartner(models.Model):
         compute_sudo=True,
     )
 
-    @api.depends("mobile", "company_id")
+    @api.depends(lambda x: x._get_same_mobile_depends())
     def _compute_same_mobile_partner_id(self):
+        empty_recordset = self.env[self._name]
```

Comment:
Isn't this usually done rather with `self.browse([])` ? Anyway it shouldn't matter